File: 49-group-anagrams/49-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        if(strs==null || strs.length==0) return new ArrayList<>();
        Map<String, List<String>> hm = new HashMap<>();
        
        for(String str : strs){
            char[] sArray = str.toCharArray();
            Arrays.sort(sArray);
            String key = new String(sArray);
            List<String> curList = hm.getOrDefault(key, new ArrayList<>());
            curList.add(str);
            hm.put(key, curList);
        }
        return new ArrayList<>(hm.values());
    }
}

File: 49-group-anagrams/49-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        anagramGroups = defaultdict(lambda:[])
        for string in strs:
            k = sorted(string)
            key = str(k)
            anagramGroups[key].append(string)
            
        res = []
        for key in anagramGroups:
            res.append(anagramGroups[key])
            
        return res
            